staging/src/k8s.io/client-go/examples/controller-ref-manager/controller/ref_manager.go

```diff
@@ -0,0 +1,142 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+
+	apps "k8s.io/api/apps/v1beta2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/kubernetes"
+	basecontroller "k8s.io/client-go/tools/controller"
+)
+
+// MyController is simple controller mock
+type MyController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+// DeploymentControllerRefManager manage deployments owner references
+type DeploymentControllerRefManager struct {
+	basecontroller.BaseControllerRefManager
+	controllerKind schema.GroupVersionKind
+	client         kubernetes.Interface
+}
+
+// NewDeploymentControllerRefManager returns a DeploymentControllerRefManager that
+// exposes methods to manage the controllerRef of deployments.
+//
+// The CanAdopt() function can be used to perform check prior to the first adoption.
+// If CanAdopt() returns a non-nil error, all adoptions will fail.
+//
+// NOTE: Once CanAdopt() is called, it will not be called again by the same
+//       DeploymentControllerRefManager instance.
+func NewDeploymentControllerRefManager(
+	client kubernetes.Interface,
+	controller metav1.Object,
+	selector labels.Selector,
+	controllerKind schema.GroupVersionKind,
+	canAdopt func() error,
+) *DeploymentControllerRefManager {
+	return &DeploymentControllerRefManager{
+		BaseControllerRefManager: basecontroller.BaseControllerRefManager{
+			Controller:   controller,
+			Selector:     selector,
+			CanAdoptFunc: canAdopt,
+		},
+		controllerKind: controllerKind,
+		client:         client,
+	}
+}
+
+// ClaimDeployments tries to take ownership of a list of Deployments.
+//
+// It will reconcile the following:
+//   * Adopt orphans if the selector matches.
+//   * Release owned objects if the selector no longer matches.
+//
+// A non-nil error is returned if some form of reconciliation was attempted and
+// failed.
+//
+// If the error is nil, either the reconciliation succeeded, or no
+// reconciliation was necessary. The list of Deployments that you now own is
+// returned.
+func (m *DeploymentControllerRefManager) ClaimDeployments(
+	deployments []*apps.Deployment) ([]*apps.Deployment, error) {
+
+	var claimed []*apps.Deployment
+	var errlist []error
+
+	match := func(obj metav1.Object) bool {
+		return m.Selector.Matches(labels.Set(obj.GetLabels()))
+	}
+	adopt := func(obj metav1.Object) error {
+		return m.adoptDeployment(obj.(*apps.Deployment))
+	}
+	release := func(obj metav1.Object) error {
+		return m.releaseDeployment(obj.(*apps.Deployment))
+	}
+
+	for _, rs := range deployments {
```

Comment:
I don't think you should claim RS because they are owned by Deployment controller - seems like you would be just fighting with it

just `return m.ClaimObject(deployment, match, adopt, release)`

Reply:
Is not RS its just my copy-paste typo.